### ISSUE TITLE
About the reproducibility of the experiment and the FID Cache issue

### ISSUE THREAD

**Issue by rookiewantstronger** [AndyCao1125/SDDPM#15]
Thanks very much to the author for open-sourcing the code.However, I encountered problems when reproducing the results of the CIFAR10 dataset.The experimental results are as follows：IS: 5.735490322113037, IS_std: 0.06943061947822571, FID: 46.21392822265625，which differs somewhat from the results in the paper.Did you make any other settings besides the hyperparameters you set in your paper?In addition, do you have any other FID CHACHE datasets?
thank u!

**Comment by hannxue** [comment]
> Thanks very much to the author for open-sourcing the code.However, I encountered problems when reproducing the results of the CIFAR10 dataset.The experimental results are as follows：IS: 5.735490322113037, IS_std: 0.06943061947822571, FID: 46.21392822265625，which differs somewhat from the results in the paper.Did you make any other settings besides the hyperparameters you set in your paper?In addition, do you have any other FID CHACHE datasets? thank u!

Hello, may I discuss paper replication with you

**Comment by hannxue** [comment]
> Of course, but I have only successfully replicated the CIFAR10 dataset.
> 
> > > Thanks very much to the author for open-sourcing the code.However, I encountered problems when reproducing the results of the CIFAR10 dataset.The experimental results are as follows：IS: 5.735490322113037, IS_std: 0.06943061947822571, FID: 46.21392822265625，which differs somewhat from the results in the paper.Did you make any other settings besides the hyperparameters you set in your paper?In addition, do you have any other FID CHACHE datasets? thank u!
> > 
> > 
> > Hello, may I discuss paper replication with you



> Of course, but I have only successfully replicated the CIFAR10 dataset.
> 
> > > Thanks very much to the author for open-sourcing the code.However, I encountered problems when reproducing the results of the CIFAR10 dataset.The experimental results are as follows：IS: 5.735490322113037, IS_std: 0.06943061947822571, FID: 46.21392822265625，which differs somewhat from the results in the paper.Did you make any other settings besides the hyperparameters you set in your paper?In addition, do you have any other FID CHACHE datasets? thank u!
> > 
> > 
> > Hello, may I discuss paper replication with you

I tried MNIST, but IS is only 1.113. I don't know why my FID value is nan

**Comment by rookiewantstronger** [comment]
> > Of course, but I have only successfully replicated the CIFAR10 dataset.
> > > > Thanks very much to the author for open-sourcing the code.However, I encountered problems when reproducing the results of the CIFAR10 dataset.The experimental results are as follows：IS: 5.735490322113037, IS_std: 0.06943061947822571, FID: 46.21392822265625，which differs somewhat from the results in the paper.Did you make any other settings besides the hyperparameters you set in your paper?In addition, do you have any other FID CHACHE datasets? thank u!
> > > 
> > > 
> > > Hello, may I discuss paper replication with you
> 
> > Of course, but I have only successfully replicated the CIFAR10 dataset.
> > > > Thanks very much to the author for open-sourcing the code.However, I encountered problems when reproducing the results of the CIFAR10 dataset.The experimental results are as follows：IS: 5.735490322113037, IS_std: 0.06943061947822571, FID: 46.21392822265625，which differs somewhat from the results in the paper.Did you make any other settings besides the hyperparameters you set in your paper?In addition, do you have any other FID CHACHE datasets? thank u!
> > > 
> > > 
> > > Hello, may I discuss paper replication with you
> 
> I tried MNIST, but IS is only 1.113. I don't know why my FID value is nan

This may be because your ‘--num_images’ parameter is set too low. The FID metric requires at least 10k generated images.

**Comment by AndyCao1125** [comment]
@rookiewantstronger 👍 

Thank you very much for your question!

A few months ago, there were also people encountering the problem of high FID values that occurred repeatedly. We conducted in-depth discussions and experiments on this issue. The specific discussion can be referred to #13 

Several factors can lead to discrepancies when using the default settings, including differences in Spikingjelly version and GPU. 

You can first take a look at this discussion. If you have any other topics you would like to discuss, please feel free to do so.

**Comment by AndyCao1125** [comment]
Regarding the other FID statistics, you can check the files in [this link](https://github.com/LuChengTHU/dpm-solver/tree/main/examples/ddpm_and_guided-diffusion).

**Comment by rookiewantstronger** [comment]
@AndyCao1125 Thank you for your reply sincerely! It has been very helpful. Additionally, could you please provide some reference for the default settings for different datasets? Futhermore,I noticed that you used the DP framework in the source code, but for multi-card training, it seems that sampling only concatenates images from one card.

**Comment by rookiewantstronger** [comment]
@AndyCao1125 Thank you very much for your patience and  wish you all the best！
